### PR TITLE
[Refactor] Refactor vectorized scan node

### LINKED ISSUE #9965: [Enhancement] Refactor vectorized scan node
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, vectorized scan nodes are inherited from its corresponding row-based scan node (e.g. VOlapScanNode <- OlapScanNode). To make codes better readability, I'd like to refactor these nodes to inherit directly from ScanNode.

Related nodes are below:

1. VOlapScanNode
2. VEsHttpScanNode
3. VMysqlScanNode
4. VOdbcScanNode
5. VSchemaScanNode
6. VBrokerScanNode

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #9965 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vbroker_scan_node.cpp [hunk 1]
```diff
@@ -26,107 +26,168 @@
...
+    RETURN_IF_CANCELLED(state);
+
+    RETURN_IF_ERROR(start_scanners());
+
+    return Status::OK();
 }
 
 Status VBrokerScanNode::start_scanners() {
     {
-        std::unique_lock<std::mutex> l(_batch_queue_lock);
```
Comment (carlvinhust2012) [inline be/src/vec/exec/vbroker_scan_node.cpp:41]: Is it necessary to change the variable name like this?

File: be/src/vec/exec/vbroker_scan_node.cpp [hunk 2]
```diff
@@ -26,107 +26,168 @@
...
-        _num_running_scanners = 1;
+        std::unique_lock<std::mutex> l(batch_queue_lock_);
```
Comment (carlvinhust2012) [inline be/src/vec/exec/vbroker_scan_node.cpp:102]: Is it necessary to change the variable name like this?

File: be/src/vec/exec/vbroker_scan_node.h [hunk 3]
```diff
@@ -43,15 +54,64 @@ class VBrokerScanNode final : public BrokerScanNode {
...
+
+    Status start_scanners();
 
     void scanner_worker(int start_idx, int length);
     // Scan one range
     Status scanner_scan(const TBrokerScanRange& scan_range, ScannerCounter* counter);
 
-    std::deque<std::shared_ptr<vectorized::Block>> _block_queue;
-    std::unique_ptr<MutableBlock> _mutable_block;
+    TupleId tuple_id_;
```
Comment (carlvinhust2012) [inline be/src/vec/exec/vbroker_scan_node.h:83]: The member variables here may be repeated to "broker_scan_node.h".

File: be/src/vec/exec/ves_http_scan_node.cpp [hunk 4]
```diff
@@ -30,68 +35,228 @@ namespace doris::vectorized {
...
+          tuple_id_(tnode.es_scan_node.tuple_id),
+          runtime_state_(nullptr),
+          tuple_desc_(nullptr),
+          num_running_scanners_(0),
+          scan_finished_(false),
+          eos_(false),
+          max_buffered_batches_(1024),
+          wait_scanner_timer_(nullptr) {}
+
+Status VEsHttpScanNode::init(const TPlanNode& tnode, RuntimeState* state) {
```
Comment (carlvinhust2012) [inline be/src/vec/exec/ves_http_scan_node.cpp:48]: This file has a lot of similar code to 'es_http_scan_node.cpp'.

File: be/src/vec/exec/vbroker_scan_node.cpp (hunk 2 above)
Comment (Gabriel39) [reply be/src/vec/exec/vbroker_scan_node.cpp:102]: I think we have discussed this convention in WeChat group. I don't plan to modify all variables now but it's better to use this convention in future

File: be/src/vec/exec/vbroker_scan_node.h (hunk 3 above)
Comment (Gabriel39) [reply be/src/vec/exec/vbroker_scan_node.h:83]: This node inherits from scan node and is independent from BrokerScanNode

File: be/src/vec/exec/ves_http_scan_node.cpp (hunk 4 above)
Comment (Gabriel39) [reply be/src/vec/exec/ves_http_scan_node.cpp:48]: Hi @carlvinhust2012 , thanks for you reviewing. As described in issue #9965 , this PR is just for refactoring. Futhermore, this PR make all scan nodes on vectorized independent from row-based ones so we could remove all row-based codes in the near future. You could just refer to this issue for motivations for this PR.

**Review by yiguolei — APPROVED** [review]
LGTM
Thanks for your greate work, we could split vectorized and non-vectorized in the future.

**Pull request merged**